pkg/resources/schema.go

```diff
@@ -213,88 +213,81 @@ func CreateContextSchema(ctx context.Context, d *schema.ResourceData, meta any)
 	database := d.Get("database").(string)
 	id := sdk.NewDatabaseObjectIdentifier(database, name)
 
-	create := true
-
 	if strings.EqualFold(strings.TrimSpace(name), "PUBLIC") {
 		_, err := client.Schemas.ShowByID(ctx, id)
-		if err != nil {
-			if !errors.Is(err, sdk.ErrObjectNotFound) {
-				return diag.FromErr(err)
-			}
-			// there is no PUBLIC schema, we need to create it
-		} else {
+		if err != nil && !errors.Is(err, sdk.ErrObjectNotFound) {
+			return diag.FromErr(err)
+		} else if err == nil {
 			// there is already a PUBLIC schema, so we need to alter it instead
-			create = false
+			log.Printf("[DEBUG] found PUBLIC schema during creation, updating...")
+			d.SetId(helpers.EncodeSnowflakeID(database, name))
+			return UpdateContextSchema(ctx, d, meta)
 		}
 	}
-	if create {
-		dataRetentionTimeInDays,
-			maxDataExtensionTimeInDays,
-			externalVolume,
-			catalog,
-			replaceInvalidCharacters,
-			defaultDDLCollation,
-			storageSerializationPolicy,
-			logLevel,
-			traceLevel,
-			suspendTaskAfterNumFailures,
-			taskAutoRetryAttempts,
-			userTaskManagedInitialWarehouseSize,
-			userTaskTimeoutMs,
-			userTaskMinimumTriggerIntervalInSeconds,
-			quotedIdentifiersIgnoreCase,
-			enableConsoleOutput,
-			err := GetAllDatabaseParameters(d)
+	// there is no PUBLIC schema, so we continue with creating
```

Comment:
nit: or the name is completely different than PUBLIC, we will handle this in the next PR